projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php

```diff
@@ -140,6 +140,35 @@ public function load_assets_with_parameters( $path_prefix, $plugin_base_path ) {
 		wp_set_script_translations( 'jetpack-instant-search', 'jetpack' );
 		$this->load_and_initialize_tracks();
 		$this->inject_javascript_options();
+
+		// It only inline the translations for the script, but does not load it.
+		$this->inject_translation_for_script(
+			plugins_url(
+				$path_prefix . '_inc/build/instant-search/jp-search.chunk-main-payload.min.js',
+				$plugin_base_path
+			)
+		);
+	}
+
+	/**
+	 * Add inline translations for script `$payload_url` before loading `$before_handle` script.
+	 *
+	 * @param string $payload_url - The payload url for which we load the translations.
+	 * @param string $before_handle - Inline the translations before this handle.
+	 */
+	protected function inject_translation_for_script( $payload_url, $before_handle = 'jetpack-instant-search' ) {
+		global $wp_version;
+		// Set a random name for the script.
+		$handle = 'jetpack-instant-search-' . wp_unique_id();
+		// Then register it, which is required for the next steps.
+		wp_register_script( $handle, $payload_url, array(), $wp_version, false );
+		// Set translation domain to `jetpack`, and we need to explicitly set the `path` to load translations files for WPCOM.
+		// Otherwise WPCOM would try to load from `WP_LANG_DIR . '/mu-plugins'` and fails.
+		wp_set_script_translations( $handle, 'jetpack', WP_LANG_DIR . '/plugins' );
+		// Inline the translations before `$before_handle` handle.
+		wp_add_inline_script( $before_handle, wp_scripts()->print_translations( $handle, false ), 'before' );
+		// Deregister the script as we don't really enqueue it from PHP side.
+		wp_deregister_script( $handle );
 	}
```

Comment:
Manually inspecting `script#jetpack-instant-search-js-before` in the document body yields a blank translation payload:

<img width="620" alt="Screen Shot 2021-09-07 at 6 33 10 PM" src="https://user-images.githubusercontent.com/4044428/132426934-6efcfbe6-f139-4f20-8d10-b4782326b999.png">

Is this expected?

Reply:
Yes this is expected. As there's no translation available for now. To simulate the situation, you'd need to 
- Change the translation file name from `jetpack-fr_FR-8df5774da753f9be21ced70eecc80000.json` to `jetpack-fr_FR-0e17ef273599d7b1004fc0c6aacf4f5e.json`. The example is done for French language, and other languages are similar, just change the hash.